From c498743f4bcfc99ff3e3e50fd768e6f4685194fe Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Thu, 25 Jul 2024 23:07:14 +0530
Subject: [PATCH] firmware: scm: Update ipq fuse list API call

Add a change to make ipq fuse list api call as generic across the platform.

Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Change-Id: I3816270f6a4bc4beacada5b2994154a3a166eaa9
---
 drivers/firmware/qcom_scm.c            | 6 +++---
 include/linux/firmware/qcom/qcom_scm.h | 9 ++++++++-
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index e4def6e7684e..ecbbfcdb99b3 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -2488,14 +2488,14 @@ int qcom_fuseipq_scm_call(u32 svc_id, u32 cmd_id,void *cmd_buf, size_t size)
 EXPORT_SYMBOL_GPL(qcom_fuseipq_scm_call);
 
 /**
- * qcom_scm_get_ipq5332_fuse_list() - Get OEM Fuse parameter from TME-L
+ * qcom_scm_get_ipq_fuse_list() - Get OEM Fuse parameter from TME-L
  *
  * @fuse: QFPROM CORR addresses
  * @size: size of fuse structure
  *
  * This function can be used to get the OEM Fuse parameters from TME-L.
  */
-int qcom_scm_get_ipq5332_fuse_list(struct fuse_payload *fuse, size_t size)
+int qcom_scm_get_ipq_fuse_list(void *fuse, size_t size)
 {
 	int ret;
 	dma_addr_t dma_fuse;
@@ -2528,7 +2528,7 @@ int qcom_scm_get_ipq5332_fuse_list(struct fuse_payload *fuse, size_t size)
 	return ret ? : res.result[0];
 
 }
-EXPORT_SYMBOL_GPL(qcom_scm_get_ipq5332_fuse_list);
+EXPORT_SYMBOL_GPL(qcom_scm_get_ipq_fuse_list);
 
 /**
  * qcom_scm_sec_auth_available() - Checks if SEC_AUTH is supported.
diff --git a/include/linux/firmware/qcom/qcom_scm.h b/include/linux/firmware/qcom/qcom_scm.h
index 2ef66f87191b..8de60d798cee 100644
--- a/include/linux/firmware/qcom/qcom_scm.h
+++ b/include/linux/firmware/qcom/qcom_scm.h
@@ -34,12 +34,19 @@
 #define QCOM_BREAK_Q6			0x1
 
 #define MAX_FUSE_ADDR_SIZE		0x8
+#define IPQ9574_MAX_FUSE_ADDR_SIZE	22
+
 struct fuse_payload {
 	uint32_t fuse_addr;
 	uint32_t lsb_val;
 	uint32_t msb_val;
 };
 
+struct fuse_payload_ipq9574 {
+	uint32_t fuse_addr;
+	uint32_t val;
+};
+
 enum qseecom_qceos_cmd_id {
 	QSEOS_APP_START_COMMAND = 0x01,
 	QSEOS_APP_SHUTDOWN_COMMAND,
@@ -323,7 +330,7 @@ extern int qcom_scm_get_device_provision_response(void *provreq_buf,
 				u32 *prov_resp_size);
 
 extern bool qcom_scm_sec_auth_available(unsigned int scm_cmd_id);
-extern int qcom_scm_get_ipq5332_fuse_list(struct fuse_payload *fuse, size_t size);
+extern int qcom_scm_get_ipq_fuse_list(void *fuse, size_t size);
 extern int qcom_sec_upgrade_auth(unsigned int scm_cmd_id,
 				 unsigned int sw_type, unsigned int img_size,
 				 unsigned int load_addr);
-- 
2.34.1

